src/transactions/test/SorobanTxTestUtils.cpp

```diff
@@ -791,48 +803,27 @@ AssetContractInvocationTest::transfer(TestAccount& fromAcc,
     auto preTransferFromBalance = mAsset.type() == ASSET_TYPE_NATIVE
                                       ? fromAcc.getBalance()
                                       : fromAcc.getTrustlineBalance(mAsset);
-    auto preTransferToBalance =
-        toAddr.type() == SC_ADDRESS_TYPE_ACCOUNT
-            ? getBalance(*mApp, toAddr.accountId(), mAsset)
-            : getContractBalance(*mApp, mContractID, toVal);
+    auto preTransferToBalance = getBalance(toAddr);
 
     // submit operation
     auto tx = sorobanTransactionFrameFromOps(mApp->getNetworkID(), fromAcc,
                                              {transfer}, {}, resources, 100,
                                              DEFAULT_TEST_RESOURCE_FEE);
-
+    invokeTx(tx, expectSuccess, false);
```

Comment:
This is slightly different than what you had before, since `invokeTx` doesn't call `tx->checkValid`. For the same behavior as before:
```
checkValid(tx);
invokeTx(tx, expectSuccess, false);
```

Reply:
Ah good point. Is there any reason why invokeTx can't call check valid? Is it used with invalid transactions?

Reply:
Check valid has some side effects, like calculating soroban resource fees for the TX. For some tests, like "basic contract invocation", we need to call check valid, initiate surge pricing, then call invoke. I don't know if calling check valid twice has any negative side effects though, so we might be able to add it to `invokeTx` but still leave it as a separate function for tests like "basic contract invocation".

Reply:
Looks like adding it to invokeTx worked

Reply:
Cool, we should probably remove all the calls to `txCheckValid` that occur right before `invokeTx` with this change.

Reply:
done

Reply:
> Check valid has some side effects, like calculating soroban resource fees for the TX.

FWIW that's no longer true. `checkValid` should not have any side effects now.